src/components/NavBar/NavBar.jsx

```diff
@@ -1,43 +1,31 @@
-import React from "react";
+import React, { useState } from "react";
 import { NavBarRightSide } from "./NavBarRightSide";
-import { NavBarContainer } from "./NavBarContainer";
+import { NavBarContainer } from "./NavBarWrapper";
 import { AvatarButton } from "./AvatarButton";
 import { UserNameLabel } from "./UsernameLabel";
 import { NavBarMenu } from "./NavBarMenu";
 import { storageService } from "services";
 
-export class NavBar extends React.Component {
-  state = {
-    anchorEl: null
-  };
+export const NavBar = ({ authContext: { onLogout } }) => {
+  const [anchorEl, updateAnchorEl] = useState(null);
 
-  handleClick = event => {
-    this.setState({ anchorEl: event.currentTarget });
-  };
+  const handleClick = ({ currentTarget }) => updateAnchorEl(currentTarget);
 
-  handleClose = () => {
-    this.setState({ anchorEl: null });
-  };
+  const handleClose = () => updateAnchorEl(null);
 
-  render() {
-    const { anchorEl } = this.state;
-    const {
-      authContext: { onLogout }
-    } = this.props;
-    const username = storageService.getUserName();
+  const username = storageService.getUserName();
```

Comment:
maybe we could have a storage hook:

```js
const [username, updateUsername] = useStorage('username');
```

what do you think?

Reply:
Yes, it's a good idea. less requests to the server (since the polling system makes a request every 1.5 secs)